.ado/templates/android-dep-setup.yml

```diff
@@ -0,0 +1,40 @@
+parameters:
+  # latest gradlew use this NDK version, which is not avaialble with any recent agents in azure
+- name: ndkVersion
+  type: string
+  default: '21.0.6113669'
+steps:
+  - task: Bash@3
+    displayName: 'NDK Download'
+    inputs:
+      targetType: 'inline'
+      # install a compatible version of NDK for gradle
+      script: echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install 'ndk;${{ parameters.ndkVersion }}'
+  
+  - task: Bash@3
+    displayName: 'Android Emulator Setup'
+    inputs:
+      targetType: 'inline'
+      script: |
+        #!/usr/bin/env bash
+        
+        # Install AVD files
+        echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install 'system-images;android-27;google_apis;x86'
+        
+        # Create emulator
+        echo "no" | $ANDROID_HOME/tools/bin/avdmanager create avd -n android_emulator -k 'system-images;android-27;google_apis;x86' --force
+        
+         # list emulator
+        echo "Available emulator"
+        $ANDROID_HOME/emulator/emulator -list-avds
```

Comment:
Why do we need to list the emulator? To make it easier to debug when you're looking at logs of previous runs?

Reply:
just incase pipeline fails it is can be tracked , if emulator was installed well.
As in next step we wait for emulator to start and up running before building